test/jdk/java/awt/TextComponent/BackgroundTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 1999, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4258667 4405602
+ * @summary  Make sure TextComponents are grayed out when non-editable
+ *           if the background color has not been set by client code.
+ *           Make sure TextComponents are not grayed out when non-editable
+ *           if the background color has been set by client code.
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual BackgroundTest
+ */
+
+import java.awt.Button;
+import java.awt.Color;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.TextArea;
+import java.awt.TextField;
+
+public class BackgroundTest {
+    private static final String enableString = "EnableText";
+    private static final String disableString = "DisableText";
+
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+                1. When the frame appears, it should have a blue background.
+                2. The first TextField and TextArea will be the default color.
+                   The second TextField and TextArea will be green.
+                4. Press the "DisableText" button.
+                   The first TextField and TextArea should change colors to the
+                   default disabled color.  On Windows, this is usually gray.
+                   On linux and macos it will match the environment settings.
+                   If the TextField or the TextArea do not change colors as described,
+                   the test FAILS.
```

Comment:
```suggestion
                3. Press the "DisableText" button.
                   The first TextField and TextArea should change colors to the
                   default disabled color.  On Windows, this is usually gray.
                   On linux and macos it will match the environment settings.
                   If the TextField or the TextArea do not change colors as described,
                   the test FAILS.
```

The instructions jump from step #2 to #4 when I was following the test. The subsequent numbers need to be adjusted as well. Test itself passes locally for me though.

Reply:
Updated.